ui/src/app/cohort-search/list-overview/list-overview.component.tsx

```diff
@@ -0,0 +1,426 @@
+import {TooltipTrigger} from 'app/components/popups';
+import {Spinner} from 'app/components/spinners';
+import {cohortBuilderApi, cohortsApi} from 'app/services/swagger-fetch-clients';
+import {reactStyles, ReactWrapperBase, withCurrentWorkspace} from 'app/utils';
+import {
+  currentCohortStore,
+  currentWorkspaceStore,
+  navigate,
+  navigateByUrl,
+  urlParamsStore
+} from 'app/utils/navigation';
+import {ConfirmDeleteModal} from 'app/views/confirm-delete-modal';
+import {Cohort, TemporalTime} from 'generated/fetch';
+import {fromJS} from 'immutable';
+import {Menu} from 'primereact/menu';
+import * as React from 'react';
+
+const COHORT_TYPE = 'AoU_Discover';
+
+const styles = reactStyles({
+  overviewHeader: {
+    display: 'flex',
+    flexFlow: 'row wrap',
+    alignItems: 'center',
+    justifyContent: 'space-between',
+  },
+  saveButton: {
+    height: '1.5rem',
+    fontSize: '12px',
+    padding: '0 0.75rem',
+    borderRadius: '3px',
+    float: 'right',
+  },
+  totalCount: {
+    marginTop: 0,
+    fontSize: '16px',
+    fontWeight: 'bold'
+  },
+  totalError: {
+    background: '#f7981c',
+    color: '#ffffff',
+    padding: '0.25rem 0.5rem',
+    borderRadius: '5px',
+    marginBottom: '0.5rem'
+  },
+  actionIcon: {
+    float: 'right',
+    margin: '0 1rem 0 0',
+    minWidth: 0,
+    padding: 0,
+    color: '#262262',
+    cursor: 'pointer'
+  },
+  disabled: {
+    opacity: 0.4,
+    cursor: 'not-allowed',
+    pointerEvents: 'none'
+  },
+  cardContainer: {
+    boxShadow: '0 0.125rem 0 0 #d7d7d7',
+    border: '1px solid #d7d7d7',
+    borderBottom: 'none',
+    borderTop: 'none',
+    borderRadius: '3px',
+  },
+  card: {
+    background: '#ffffff',
+    marginBottom: 0,
+    marginTop: 0,
+  },
+  cardHeader: {
+    color: '#302C71',
+    fontSize: '13px',
+    borderBottom: 'none',
+    padding: '0.5rem 0.75rem',
+  },
+  error: {
+    background: '#f7981c',
+    color: '#ffffff',
+    fontSize: '11px',
+    border: '1px solid #ebafa6',
+    borderRadius: '3px',
+    marginBottom: '0.25rem',
+    padding: '3px 5px'
+  },
+  invalid: {
+    background: '#f5dbd9',
+    color: '#565656',
+    fontSize: '11px',
+    border: '1px solid #ebafa6',
+    borderRadius: '3px',
+    marginBottom: '0.25rem',
+    padding: '3px 5px'
+  },
+});
+
+interface Props {
+  searchRequest: any;
+  update: number;
+}
+
+interface State {
+  error: boolean;
+  loading: boolean;
+  total: number;
+  chartData: any;
+  saveModal: boolean;
+  name: string;
+  description: string;
+  nameTouched: boolean;
+  saving: boolean;
+  deleting: boolean;
+  stackChart: boolean;
+  showConflictError: boolean;
+  saveError: boolean;
+  cohort: Cohort;
+  apiCallCheck: number;
+}
+
+export const ListOverview = withCurrentWorkspace()(
+  class extends React.Component<Props, State> {
+    dropdown: any;
+    constructor(props: Props) {
+      super(props);
+      this.state = {
+        error: false,
```

Comment:
can the name of this var be more specific?  it's referring to errors that occur when making api calls right, not input errors.  it gets confusing below with `this.hasErrors` and other error checking logic

Reply:
Good point, it is confusing. I changed `error` to `apiError` and `hasErrors` to `definitionErrors`